FirebaseStorageUI/UIImageView+FirebaseStorage.h

```diff
@@ -0,0 +1,112 @@
+//
+//  Copyright (c) 2016 Google Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+@import UIKit;
+
+@import FirebaseStorage;
+@import SDWebImage;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UIImageView (FirebaseStorage)
+
+/**
+ * The current download task, if the image view is downloading an image.
+ */
+@property (nonatomic, readonly, nullable) FIRStorageDownloadTask *sd_currentDownloadTask;
+
+/**
+ * Sets the image view's image to an image downloaded from the Firebase Storage reference.
+ *
+ * @param storageRef  A Firebase Storage reference containing an image.
+ * @return Returns a FIRStorageDownloadTask if a download was created (i.e. image
+ *   could not be found in cache).
+ */
+- (nullable FIRStorageDownloadTask *)sd_setImageWithStorageReference:(FIRStorageReference *)storageRef;
+
+/**
+ * Sets the image view's image to an image downloaded from the Firebase Storage reference.
+ *
+ * @param storageRef  A Firebase Storage reference containing an image.
+ * @param placeholder An image to display while the download is in progress.
+ * @return Returns a FIRStorageDownloadTask if a download was created (i.e. image
+ *   could not be found in cache).
+ */
+- (nullable FIRStorageDownloadTask *)sd_setImageWithStorageReference:(FIRStorageReference *)storageRef
+                                                    placeholderImage:(UIImage *)placeholder;
+
+/**
+ * Sets the image view's image to an image downloaded from the Firebase Storage reference.
+ *
+ * @param storageRef  A Firebase Storage reference containing an image.
+ * @param placeholder An image to display while the download is in progress.
+ * @param completion  A closure to handle events when the image finishes downloading.
+ *   The closure is not guaranteed to be invoked on the main thread.
```

Comment:
Devs can by immediately dispatching inside of the block, since they control the block param that's passed in. Currently this method does dispatch the completion to the main queue, I just wanted the documentation to be flexible toward changes in the future.